src/Classifiers/OneVsRest.php

```diff
@@ -0,0 +1,254 @@
+        ];
+    }
+
+    /**
+     * Has the learner been trained?
+     *
+     * @return bool
+     */
+    public function trained() : bool
+    {
+        return !empty($this->classifiers);
+    }
+
+    /**
+     * Train the learner with a dataset.
+     *
+     * @param \Rubix\ML\Datasets\Labeled $dataset
+     */
+    public function train(Dataset $dataset) : void
+    {
+        SpecificationChain::with([
+            new DatasetIsLabeled($dataset),
+            new DatasetIsNotEmpty($dataset),
+            new SamplesAreCompatibleWithEstimator($dataset, $this),
+            new LabelsAreCompatibleWithLearner($dataset, $this),
+        ])->check();
+
+        $classes = $dataset->possibleOutcomes();
+
+        $this->backend->flush();
+
+        foreach ($classes as $class) {
+            $estimator = clone $this->base;
+            $subset = clone $dataset;
+
+            $binarize = function ($label) use ($class) {
+                return $label === $class ? 'y' : 'n';
+            };
+
+            $subset->transformLabels($binarize);
+
+            $task = new TrainLearner($estimator, $subset);
+
+            $this->backend->enqueue($task);
+        }
+
+        $classifiers = $this->backend->process();
+
+        $classifiers = array_combine($classes, $classifiers) ?: [];
+
+        $this->classifiers = $classifiers;
+
+        $this->featureCount = $dataset->numFeatures();
+    }
+
+    /**
+     * Make predictions from a dataset.
+     *
+     * @param \Rubix\ML\Datasets\Dataset $dataset
+     * @throws \Rubix\ML\Exceptions\RuntimeException
+     * @return list<string>
+     */
+    public function predict(Dataset $dataset) : array
+    {
+        return array_map('\Rubix\ML\argmax', $this->proba($dataset));
+    }
+
+    /**
+     * Estimate the joint probabilities for each possible outcome.
+     *
+     * @param \Rubix\ML\Datasets\Dataset $dataset
+     * @throws \Rubix\ML\Exceptions\RuntimeException
+     * @return list<array<string,float>>
+     */
+    public function proba(Dataset $dataset) : array
+    {
+        if (!$this->classifiers or !$this->featureCount) {
+            throw new RuntimeException('Estimator has not been trained.');
+        }
+
+        DatasetHasDimensionality::with($dataset, $this->featureCount)->check();
+
+        $this->backend->flush();
+
+        /** @var \Rubix\ML\Probabilistic $estimator */
+        foreach ($this->classifiers as $estimator) {
+            $task = new Proba($estimator, $dataset);
+
+            $this->backend->enqueue($task);
+        }
+
+        $aggregate = $this->backend->process();
+
+        $aggregate = array_transpose($aggregate);
+
+        $classes = array_keys($this->classifiers);
+
+        $probabilities = [];
+
+        foreach ($aggregate as $votes) {
+            $dist = [];
+
+            foreach ($votes as $j => $proba) {
+                $dist[$classes[$j]] = $proba['y'];
+            }
+
+            $total = array_sum($dist);
```

Comment:
We could also potentially distribute the 'n' probabilities over the remaining classes so that the distribution still adds to 1 and there would be no need for this `array_sum` (O(n)) as we'd just divide over the number of binary classifiers instead.

Reply:
This actually changes the probabilities (it has a smoothing effect) and makes it less accurate in the unit test so not implementing.